Add the section to describe the field spec.version (#2697)
knative/docs: docs/install/operator/configuring-eventing-cr.md

- [Version Configuration](#version-configuration)
- [Eventing Configuration by ConfigMap](#eventing-configuration-by-configMap)
- [Private repository and private secret](#private-repository-and-private-secrets)
- [Configuring default broker class](#configuring-default-broker-class)

__NOTE:__ Kubernetes spec level policies cannot be configured using the Knative operators.

## Version Configuration

As of 0.16.0, Knative users are able to install Knative Eventing of a specific version via the field `spec.version`. For
example, if you want to install Knative Eventing 0.16.0, you can apply the following `KnativeEventing` custom resource:

```
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
spec:
  version: 0.16.0
```

If the field `spec.version` is not specified, Knative Operator will install Knative Eventing of the latest version available
in the operator. If users specify an invalid or unavailable version, Knative Operator will do nothing. Knative Operator
does not bundle Knative components with the version, 2 Minor versions or more earlier than the current operator version.

> For example, if the version of Knative operator is 0.16.x, the earliest version of Knative Eventing is 0.14.0, as available

Reviewer: Should we say something like "we'll always include 3 patch releases"?

Author: @Cynocracy Do you mean we'll always include 3 minor releases?

Reviewer: Er, yeah, sorry, mis-type. Should we commit to a window of 3 minor for now?

Author: yes.

The Knative Eventing operator can be configured with these options:

- [Version Configuration](#version-configuration)
- [Eventing Configuration by ConfigMap](#eventing-configuration-by-configMap)
- [Private repository and private secret](#private-repository-and-private-secrets)
- [Configuring default broker class](#configuring-default-broker-class)

__NOTE:__ Kubernetes spec level policies cannot be configured using the Knative operators.

## Version Configuration

> As of 0.16.0, Knative users are able to install Knative Eventing of a specific version via the field `spec.version`. For

Reviewer 2: Can we say "administrators" or "operators" -- I think we want to make clear that this is targeted at the "operator" persona rather than the "developer" persona:

![image](https://user-images.githubusercontent.com/7959095/88423182-3b34d100-cda0-11ea-806c-6da326ad5dc2.png)

Author: done.

> As of 0.16.0, Knative administrators are able to install Knative Eventing of a specific version via the field `spec.version`. For
> example, if you want to install Knative Eventing 0.16.0, you can apply the following `KnativeEventing` custom resource:

Reviewer 3: ```suggestion
Cluster administrators can install a specific version of Knative Eventing by using the `spec.version` field. For
example, if you want to install Knative Eventing 0.16.0, you can apply the following `KnativeEventing` custom resource:
```

```
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
spec:
  version: 0.16.0
```

If the field `spec.version` is not specified, Knative Operator will install Knative Eventing of the latest version available

> in the operator. If users specify an invalid or unavailable version, Knative Operator will do nothing. Knative Operator

Reviewer 2: I think if there is an invalid or unavailable version, the custom resource will signal an error in status?

Is it worth highlighting how to view the status to ensure that the correct version is operating?

Author: Not for this case I am afraid.
We need to change the operator behavior, if the version is not found.

always includes 3 minor releases. For example, if the version of Knative operator is 0.16.x, the earliest version of

> Knative Eventing is 0.14.0, as available in this operator.

Reviewer 2: Do we want to suggest [`0.14.2`](https://github.com/knative/eventing/releases/tag/v0.14.2), as that is the latest patch version of 0.14?

(Do we support automatic "latest patch" semver, or only exact match?)

Author: done.

> If the field `spec.version` is not specified, Knative Operator will install Knative Eventing of the latest version available
> in the operator. If users specify an invalid or unavailable version, Knative Operator will do nothing. Knative Operator
> always includes 3 minor releases. For example, if the version of Knative operator is 0.16.x, the earliest version of
> Knative Eventing is 0.14.2, as available in this operator.

Reviewer 3: ```suggestion
If `spec.version` is not specified, the Knative Operator will install the latest available version of Knative Eventing.
If users specify an invalid or unavailable version, the Knative Operator will do nothing.
The Knative Operator always includes the latest 3 minor release versions. For example, if the current version of the Knative Operator is 0.16.x, the earliest version of Knative Eventing available through the Operator is 0.14.2.
```

> If there has been Knative Eventing already installed by the operator, applying the `KnativeEventing` CR with a different
> `spec.version` means upgrading or downgrading the Knative Eventing. However, Knative operator only allows version migration
> from the existing version to the adjacent minor releases. For example, if the existing Knative Eventing is in 0.14.x, you
> can only upgrade to 0.15.x.

Reviewer 2: ```suggestion
If Knative Eventing is already managed by the operator, updating the `spec.version` field in the `KnativeEventing` resource
allows upgrading or downgrading the Knative Eventing version without needing to change the operator. Note that the Knative operator only allows
upgrade or downgrade by a single minor release at this time. For example, if the existing Knative Eventing is in 0.14.x, you
must upgrade to 0.15.x. before upgrading to 0.16.x.
```

Author: done

> If Knative Eventing is already managed by the operator, updating the `spec.version` field in the `KnativeEventing` resource
> allows upgrading or downgrading the Knative Eventing version without needing to change the operator. Note that the Knative operator only allows
> upgrade or downgrade by a single minor release at this time. For example, if the existing Knative Eventing is in 0.14.x, you
> must upgrade to 0.15.x, before upgrading to 0.16.x.

Reviewer 3: ```suggestion
If Knative Eventing is already managed by the Operator, updating the `spec.version` field in the `KnativeEventing` resource enables upgrading or downgrading the Knative Eventing version, without needing to change the Operator. 

Note that the Knative Operator only permits upgrades or downgrades by one minor release version at a time. For example, if the current Knative Eventing deployment is version 0.14.x, you must upgrade to 0.15.x before upgrading to 0.16.x.
```
